### PR TITLE
feat: add date-helpers package

### PR DESCRIPTION
**Pull request by liby**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Add a new package: `@rightcapital/date-helpers`

Issue Number: N/A

## What is the new behavior?
N/A

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/date-helpers/src/date-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+/**
+ * A utility class that provides various date formatting and parsing methods.
+ */
+export class DateHelpers {
+  /** Format for date strings in US locale, e.g., "12/31/2023" */
+  public static usLocaleDateFormat = 'MM/dd/yyyy';
+
+  /** Format for date and time strings in US locale, e.g., "12/31/2023 23:59:59" */
+  public static usLocaleTimeFormat = 'MM/dd/yyyy HH:mm:ss';
```
Comment (lixiaoyan) [inline packages/date-helpers/src/date-helpers.ts:20]:
```suggestion
  public static usLocaleDateTimeFormat = 'MM/dd/yyyy HH:mm:ss';
```

File: packages/date-helpers/src/date-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+   *
+   * const usTimeString = DateHelpers.formatUsLocaleTime("2025/10/20 12:00:00");
+   * // Result: "10/20/2025 12:00:00"
+   * ```
+   *
+   * @param dateSource - The date to format, can be a Date object or a string.
+   * @param dateFormat - Optional, a custom date format string.
+   * @returns The formatted time string.
+   */
+  public static formatUsLocaleTime(
```
Comment (lixiaoyan) [inline packages/date-helpers/src/date-helpers.ts:158]:
```suggestion
  public static formatUsLocaleDateTime(
```

File: packages/date-helpers/src/date-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    if (dateSource === '') {
+      return dateSource;
+    }
+
+    if (dateFormat) {
+      return format(getParsedDate(dateSource), dateFormat);
+    }
+
+    return format(getParsedDate(dateSource), DateHelpers.usLocaleDateFormat);
+  }
```
Comment (lixiaoyan) [inline packages/date-helpers/src/date-helpers.ts:138]: Rename this method to `format` and make the second parameter mandatory, then use this method for all other format methods.

File: packages/date-helpers/src/date-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+   * @param dateSource - The date to format, can be a Date object or a string.
+   * @returns The formatted date string.
+   *
+   * @remarks Naming Origin: Derived from the Angular DatePipe official documentation.
+   *
+   * Reference:
+   *
+   * https://angular.io/api/common/DatePipe#:~:text=mediumDate
+   */
+  public static formatMediumDate(dateSource: DateType): string {
```
Comment (lixiaoyan) [inline packages/date-helpers/src/date-helpers.ts:191]: nit: Should we add `us locale` in the method name or remove it from others for consistency?

**Review by lixiaoyan — CHANGES_REQUESTED** [review]

File: packages/date-helpers/src/date-helpers.ts (hunk 2 above)
Comment (rainx) [reply packages/date-helpers/src/date-helpers.ts:158]: formatDateToUSLocaleDateTimeString ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString

Comment (rainx) [reply packages/date-helpers/src/date-helpers.ts:158]: er... `formatDateToXxx` is also not a good practice. our source param is not Date.   |o|

**Review by lixiaoyan — APPROVED** [review]

**Review by rainx — APPROVED** [review]
There are some comments that need to be updated. please update them in the next PR.

**Pull request merged**